### DISCUSSION TITLE
コマンドの引数の順番

### DISCUSSION THREAD

**Discussion by KEINOS** [Qithub-BOT/QiiCipher#36] [一般]
以下が v1.0.0 f7e34f9 現在のコマンドの引数一覧です。

| コマンド | 第 1 引数 | 第 2 引数 | 第 3 引数 | 第 4 引数 |
| :--:| :--: | :--: | :--: | :--: |
| `archive` | `<input file>` | [`<output file>`] |
| `check` | `<github user>` | `<id_rsa>` | |
| `checkkeylength` | `<github user>` | 
| `dec` | `<private key>` |  `<input file>` |  `<output file>` |
| `enc` | `<github user>` | `<input file>` |  [`<output file>`] |
| `keygen` | `<email>` | `<keyname>` | 
| `sign` | `<github user>` | `<private key>` | `<input file>` | [`<output file>`] |
| `verify` | `<verify file>` | `<github user>` | [`<sign file>`] |

コマンドを動詞、第 1 引数を対象というのがベースにあります。

しかし、`sign` で署名してから `verify` で確認しようとすると、いささか引数の感覚がチャンポン気味になってしまいます。

引数の順番は v2.0.0 で変更するとして、みなさんの意見を聞きたいです。

- 関連 Issue https://github.com/Qithub-BOT/QiiCipher/issues/31#issuecomment-852686386

**Comment by yoshi389111** [comment]
既存のコマンドの引数順番の多数決をとる（統一感を出す）と、大体以下のようになる感じですかね。

1. `<github user>`
2. `<private key>`
3. `<input file>`
4. `<output file>`

（不要なものは削除、複数ある場合もあり）

上記の順番と違うものは `keygen` と `verify` の2つだけかな？

`keygen` は引数の毛色が違う気がするので、結果的に `verify` の順番だけ違和感があるってことなんですかね。

`verify` も統一感をだすなら、以下のような感じ？

`verify <github user> <verify file> [<sign file>]`

**Comment by KEINOS** [comment]
v2.0 の話しなので可能性の話しとして。

- 基本的に「第 1 引数のみ」に限定して、残りはオプションで指定する

```bash
verify <verify file> --user <github user> --sigfile <sign file>
verify <verify file> -u <github user> -s <sign file>
```

[Qiitadon の LTL で知った](https://qiitadon.com/web/statuses/106338255953727190)のですが、POSIX で動くライブラリがあるようです。（オプション実装チャレンジしたけど、めんどいので）

- getoptions https://github.com/ko1nksm/getoptions @ GitHub
